src/tools/rbd_mirror/PoolWatcher.cc

```diff
@@ -95,6 +95,9 @@ void PoolWatcher<I>::init(Context *on_finish) {
     m_on_init_finish = on_finish;
   }
 
+  assert(!m_refresh_in_progress);
+  m_refresh_in_progress = true;
```

Comment:
@dillaman I am just wondering why do you access m_refresh_in_progress not holding the lock (here and inprocess_refresh_images below)?